### DISCUSSION TITLE
Expose store context

### DISCUSSION THREAD

**Discussion by ritchieanesco** [vincjo/svelte-simple-datatables#48] [Ideas]
Great library @vincjo!

I was wondering if there was a possibility of exposing the store context? This will allow the lib to be more flexible so that users can create their own filters and search field components rather than using the in-built components.

svelte-form-lib has a pretty example of doing this https://svelte-forms-lib-sapper-docs.vercel.app/custom. Let me know what you think?

**Comment by vincjo** [comment]
Hello ritchieanesco,

That's the next step, I agree with you.

I intend to refactor the code, especially on all the part that manipulates the DOM, and switch to typescript. Once realized, it will be much easier to create an expressive and easy-to-use lib

Thank you very much for the link, I'll take a look at it 👍

**Comment by vincjo** [comment]
I created a separate lib which will be integrated into the Datatable component, later.
https://vincjo.fr/svelte-data-handler
